pkg/broadcaster/broadcaster.go

```diff
@@ -0,0 +1,77 @@
+package broadcaster
+
+import (
+	"errors"
+	"io"
+	"sync"
+)
+
+// Broadcaster is a simplified Broker allowing clients to subscribe while
+// itself behaving as a io.Writer.
+type Broadcaster struct {
+	mu sync.Mutex
+	// NOTE: As we use a map, the broadcast order is randomized.
+	// Consider using a slice instead to be deterministic.
+	clients map[io.WriteCloser]struct{}
```

Comment:
The map is used because we want `clients==nil` to mean "closed", correct?

Reply:
That helps, but mostly for evict/unsubscribe, calling `delete(map, key)` is easier/cleaner than to do slice operation by appenning subslices and messing up indexes.